### PR TITLE
Update battle.jl

### PR DESCRIPTION
**Pull request by BonelessPi**
* Pass ifempty=false to walk!() calls so that agents can start to fight
* Rename battle() to create_battlefield() to better match the purpose of the function. Also helps avoid confusion with the unrelated function battle!()

### CODE REVIEW

**Comment by BonelessPi** [comment]
In response to https://github.com/JuliaDynamics/AgentsExampleZoo.jl/issues/23

**Comment by Datseris** [comment]
thanks. i assume you've run the example and the video works fine?

File: docs/examples/battle.jl [hunk 1]
```diff
@@ -182,7 +182,7 @@ function captor_behavior!(agent, model)
         ## Prisoner runs away in the commotion
         prisoner.shape = :utriangle
         prisoner.capture_time = 0
-        walk!(prisoner, (rand(abmrng(model), -1:1, 2)..., 0), model)
+        walk!(prisoner, (rand(abmrng(model), -1:1, 2)..., 0), model, ifempty=false)
```
Comment (Datseris) [inline docs/examples/battle.jl]:
```suggestion
        walk!(prisoner, (rand(abmrng(model), -1:1, 2)..., 0), model; ifempty=false)
```

i believe it makes for a more readable code if keywords are separated by `;` can you change this here and in subsequent calls?

**Comment by BonelessPi** [comment]
https://github.com/user-attachments/assets/03dc77fe-d749-466a-a70d-64f15a842321

Here's the video that it generates for me. Technically the video starts after 5 steps have already passed in the model.

File: docs/examples/battle.jl (hunk 1 above)
Comment (BonelessPi) [reply docs/examples/battle.jl]: Added this change in new commit

**Pull request merged**